extension/encoding/awscloudwatchmetricstreamsencodingextension/opentelemetry_unmarshaler.go

```diff
@@ -4,15 +4,66 @@
 package awscloudwatchmetricstreamsencodingextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awscloudwatchmetricstreamsencodingextension"
 
 import (
+	"errors"
 	"fmt"
 
+	"github.com/gogo/protobuf/proto"
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awscloudwatchmetricstreamsencodingextension/internal/metadata"
 )
 
-type formatOpenTelemetry10Unmarshaler struct{}
+var errInvalidOTLPMessageLength = errors.New("unable to decode data length from message")
+
+type formatOpenTelemetry10Unmarshaler struct {
+	buildInfo component.BuildInfo
+}
 
 var _ pmetric.Unmarshaler = (*formatOpenTelemetry10Unmarshaler)(nil)
 
-func (f formatOpenTelemetry10Unmarshaler) UnmarshalMetrics(_ []byte) (pmetric.Metrics, error) {
-	return pmetric.Metrics{}, fmt.Errorf("UnmarshalMetrics unimplemented for format %q", formatOpenTelemetry10)
+// formatOpenTelemetryError appends message indicating
+// formatOpenTelemetry10 as origin to the beginning of the error
+func formatOpenTelemetryError(err error) error {
+	return fmt.Errorf("failed to unmarshal metrics as '%s' format: %w", formatOpenTelemetry10, err)
+}
```

Comment:
Perhaps instead of having this and `formatJSONError`, we should add an `UnmarshalMetrics` method to the extension type and have it call the underlying unmarshaler, and wrap the error with this message.